src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -329,8 +328,17 @@ public static IHostBuilder AddScriptHostCore(this IHostBuilder builder, ScriptAp
 
                 services.AddSingleton<IFileLoggingStatusManager, FileLoggingStatusManager>();
 
-                if (!applicationHostOptions.HasParentScope)
+                if (applicationHostOptions.HasParentScope)
+                {
+                    // Forward th host LanguageWorkerOptions to the Job Host.
+                    var languageWorkerOptions = applicationHostOptions.RootServiceProvider.GetService<IOptionsMonitor<LanguageWorkerOptions>>();
```

Comment:
I am slightly concerned about how this will work in practice. There are subtleties to the way options are registered and calculated. I would feel better if we did the following:

``` CSharp
services.AddSingleton(_ => applicationHostOptions.RootServiceProvider.GetService<IOptionsMonitor<LanguageWorkerOptions>>());
services.AddScoped(_ => applicationHostOptions.RootServiceProvider.GetService<IOptionsSnapshot<LanguageWorkerOptions>>());
services.AddSingleton(_ => applicationHostOptions.RootServiceProvider.GetService<IOptions<LanguageWorkerOptions>>());
```

Could even lift this to an extension method `IServiceCollection ForwardOptionsFrom<TOptions>(this IServiceCollection, IServiceProvider source);`

Reply:
> Also, one thing to note is that using the suggested pattern will defer execution of the setup to the time services in the child container request the options, if nothing at the host scope consumes one of those option types.

That was intentional. I wanted to ensure `IOptions` in both parent and child container would be the same values. Otherwise, the two containers could potentially have different values if config changes between the two. Unlikely, but that would be one confusing bug if it ever happens.

> we are explicitly trying to avoid another cache miss when using IOptions<T>

Would this cache miss not already exist in the parent container?

Reply:
With the pattern currently used in the code, the values would be the same, also refreshed when using `IOptionsMonitor`, one of the differences is that when using `IOptions<T>`, you'd end up with the same instance that is returned by `IOptionsMonitor<T>` (at the initial construction here), but that's intentional.

> Would this cache miss not already exist in the parent container?

Right now, there are two cache misses in the JobHost/ScriptHost scope with every single initialization:

1. Services taking a dependency on `IOptionsMonitor<LanguageWorkerOptions>`
2. Services taking a dependency on `IOptions<LanguageWorkerOptions>`

If no WebHost services are using `IOptions<LanguageWorkerOptions>` (or if that ever changes in the future as a result of changes to WebHost scoped services), we'd end up with a cache miss when initializing the JobHost, which would have an impact on cold start in some cases (specialization flows shouldn't be as impacted as the expectation is that placeholders would have forced this to happen).

The `IOptions<LanguageWorkerOptions>` is the second hit we're trying to avoid.

For the WebHost, in specialization cases, even if the host has services consuming `IOptions<LanguageWorkerOptions>`, those wouldn't impact customer cold start as they would happen in placeholders.

Ultimately, much of what we're dealing with here is the fact the this current setup implementation is doing way too much and that logic should be refactored into a separate, singleton, service that is reused, avoiding the duplicate execution of that expensive logic that is not expected to run more than once anyway.

Reply:
The WebHost won't ever be realizing `IOptions<LanguageWorkerOptions>` when we have a customer payload? If that is the case, then I am fine with the approach as is.